stable/influxdb/templates/meta-statefulset.yaml

```diff
@@ -0,0 +1,132 @@
+{{- if .Values.enterprise.enabled }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: "{{ template "influxdb.fullname" . }}-meta"
+  labels:
+    app: "{{ template "influxdb.fullname" . }}-meta"
+    chart: "{{ template "influxdb.chart" . }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  replicas: {{ .Values.enterprise.meta.clusterSize }}
+  selector:
+    matchLabels:
+      app: "{{ template "influxdb.fullname" . }}-meta"
+  serviceName: "{{ template "influxdb.fullname" . }}-meta"
+  template:
+    metadata:
+      labels:
+        app: "{{ template "influxdb.fullname" . }}-meta"
+        release: "{{ .Release.Name }}"
+      {{- if .Values.podAnnotations }}
+      annotations:
+{{ toYaml .Values.podAnnotations | indent 8 }}
+      {{- end }}
+    spec:
+      {{- if .Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range .Values.image.pullSecrets }}
+        - name: {{ . }}
+      {{- end}}
+      {{- end }}
+      containers:
+      - name: "{{ template "influxdb.fullname" . }}-meta"
+        image: "{{ .Values.image.repository }}:{{ .Values.enterprise.meta.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+        resources:
+{{ toYaml .Values.enterprise.meta.resources | indent 10 }}
+        ports:
+        - name: udp
+          containerPort: {{ .Values.config.udp.bind_address }}
+        - name: rpc
+          containerPort: {{ .Values.config.rpc.bind_address }}
+        - name: meta
+          containerPort: {{ .Values.config.meta.bind_address }}
+        {{- if .Values.env }}
+        env:
+{{ toYaml .Values.env | indent 10 }}
+          # Values.env's HOSTNAME isn't fundamentally different from $HOSTNAME, but this way we get a distinguished name for InfluxDB at runtime.
+          - name: INFLUXDB_HOSTNAME
+            value: "$(_HOSTNAME).{{ template "influxdb.fullname" . }}-meta"
```

Comment:
Not familiar with enterprise so not sure how/what this is used for, but would "{{ template "influxdb.fullname" . }}-meta-<stateful set replica number> (Which is value for $HOSTNAME) not be sufficiently distinguished?

Reply:
Sorry, "distinguished" wasn't a great fit here. I meant like a FQDN. InfluxDB Enterprise needs names to be resolvable.